src/InputParser.jl

```diff
@@ -1,18 +1,52 @@
 module InputParser
 
-# Julia implicitly numbers Enum items 0,1,2,3,4 (in order)
-# I put them in same order as FORTRAN code (i.e. NOPT=0 meant
-# ConsolidationSwell so Int(ConsolidationSwell) = 0) however
-# this wont matter since we should never be referring to the integer 
-# values explicitly anyways
+# Julia implicitly numbers Enum items 0,1,2,3,4
 @enum Model ConsolidationSwell LeonardFrost Schmertmann CollapsibleSoil SchmertmannElastic 
 @enum Foundation ErrorFoundation RectangularSlab LongStripFooting 
+@enum ErrorID ParsingErrorId FoundationErrorId SoilNumberErrorId NotEnoughValuesErrorId
+
+#### Custom Exceptions  ################################
+
+# ParsingError is for when we have already caught an error 
+# in the body of InputData constructor, given user feedback, 
+# but want to throw another error so caller of InputData 
+# constructor can decide what to do
+struct ParsingError <: Exception 
+    id::Int
+    ParsingError()=new(Int(ParsingErrorId))
+end
+Base.showerror(io::IO, e::ParsingError) = print(io, "Could not parse file.")
+
+# FoundationError is for when foundationOption is not read as 1 or 2
```

Comment:
I know you are using the same file format as the original file, but once we get this working, we should update the file format to use a more meaningful string for the input.  Remember what 1 or 2 means is asking too much of the user.  Ideally, we'll eventually have the GUI interface which would let someone pick their foundation option with a drop down.